src/components/agencies/view-agency/view-agency.component.js

```diff
@@ -30,16 +41,36 @@ class ViewAgency extends Component {
     });
   };
 
+  goEditUser = (id) => {
+    //TODO: Use router!
+    goToPage(EDIT_USER_PAGE, id);
+  };
+
   componentDidMount() {
     const { id } = this.props.match.params;
-
     this.setState({ loading: true }, () => {
       agencyService
         .getAgency(id)
         .then((data) => {
           this.setState({
             agency: data,
-            loading: false,
+          });
+
+          boardingService.getBoardingsWithFacet(50,0,null,{"agency":data.name}).then((userData)=>{
```

Comment:
If I'm reading this properly, this gets a list of officers by getting the boarding records for the agency, and finding all the distinct reportingOfficer email addresses. 

But we want to see officers, even if they have not submitted any boarding records.

This should use the "searchFacetByUsers" function, which takes in a limit, offset, query, and filter, and returns an array of users (using limit and offset to only return what's needed on the page), a total count of users, and any highlighting needed. 

You can see that function at https://github.com/WildAid/o-fish-realm/blob/main/WildAidDemo/functions/searchFacetByUsers/source.js

Reply:
I tried using getUsers from users.service which calls searchFacetByUsers function. With agency name as filter I received empty array in response & without any filters I received list of users, In my case I received user details of me & another user I created. So what parameters would be helpful to get officer details ?

Reply:
Ah, you don't need a filter, there are permissions in place to make sure you only see what you're allowed to see. (as an agency admin, you can only see users in your agency)

You say you get the user details for you and the other user you created, which sounds like you have the right link there. What additional details do you need?

Reply:
Yes I think response from getUsers will suffice the requirement. but one doubt I have is, on agencies page for agency name deveshchatuphale7 respective officer name in the list is Emex Geefy.But, I didn't receive any info about that user in the response of getUsers without any filters & searchQuery.
So just wanted to know if I am receiving a correct response or I have to pass any additional filters or searchQuery ?

Reply:
To debug, loop through every part of the returned user object, and print it, to see where the issue is. Perhaps there is a different field you should be using? (that's what I'd do, I don't know where the problem is here).